nixos/modules/config/users-groups.nix

```diff
@@ -53,37 +53,80 @@ let
         "*"    # password unset
       ]);
 
+  overrideOrderMutable = ''
+    {option}`initialHashedPassword` -> {option}`initialPassword` ->
+    {option}`hashedPassword` -> {option}`password`
+    {option}`hashedPasswordFile`
+  '';
+
+  overrideOrderImmutable = ''
+    {option}`initialHashedPassword` -> {option}`hashedPassword` ->
+    {option}`initialPassword` -> {option}`password`
+    {option}`hashedPasswordFile`
+  '';
+
+  overrideOrderText = isMutable: ''
+    If the option {option}`users.mutableUsers` is
+    `${builtins.toString isMutable}`, then the order of precedence is as shown
+    below, where values on the left are overridden by values on the right:
+
+    ${if isMutable then overrideOrderMutable else overrideOrderImmutable}
```

Comment:
I've rebased with new commits following the contribution guideline and at the same time I _think_ I addressed this suggestion, but please take a look at confirm it's what you wanted. 

I will look into how to actually generate the new docs from nixpkgs, so I can confirm things are formatted sanely.

Reply:
Hah.. there is indeed a rendering error I'll fix, but figured I'd note it down here anyway. Built with `nix-build nixos/release.nix -A manual.x86_64-linux` and checked the `options.html`:

> If the option users.mutableUsers is ``, then the order of precedence is as shown below, where values on the left are overridden by values on the right: initialHashedPassword -> hashedPassword -> initialPassword -> password -> hashedPasswordFile

> If the option users.mutableUsers is 1, then the order of precedence is as shown below, where values on the left are overridden by values on the right: initialHashedPassword -> initialPassword -> hashedPassword -> password -> hashedPasswordFile


Is because:
```
nix-repl> builtins.toString true
"1"

nix-repl> builtins.toString false
""
```

Reply:
The rendering issue is fixed now

Reply:
> Is because:

Heh, didn't even notice it during review.

Fwiw, this is (IIRC) because Nix does `toString` on all derivation attributes before putting those into the env to make them accessible for the builder (almost always sh/bash) and it kinda makes sense to have an empty string for false in such a language. But yeah, it's a surprise the first time for everybody ;-)

Reply:
Thanks for the insight! It seemed a bit cursed at first, but in the context of generating shellscript output it does indeed make a lot of sense.